lib/Frontend/nvqpp/ConvertStmt.cpp

```diff
@@ -121,9 +121,61 @@ bool QuakeBridgeVisitor::TraverseCXXCatchStmt(clang::CXXCatchStmt *x,
 }
 
 bool QuakeBridgeVisitor::TraverseCXXForRangeStmt(clang::CXXForRangeStmt *x,
-                                                 DataRecursionQueue *q) {
-  TODO_x(toLocation(x), x, mangler, "for (ranged) statement");
-  return false;
+                                                 DataRecursionQueue *) {
+  auto loc = toLocation(x);
+  if (!TraverseStmt(x->getRangeInit()))
+    return false;
+  Value buffer = popValue();
+  auto stdvecTy = dyn_cast<cc::StdvecType>(buffer.getType());
```

Comment:
What is the reason this has to be a StdvecType? Could we also extend this to `for (auto& q : my_qreg) `?

Reply:
Yes. For this PR, I was only dealing with std::vector, but we should add other iterators, etc. to the bridge.